pyproject.toml

```diff
@@ -0,0 +1,5 @@
+[project]
+requires-python = ">= 3.11"
+
+[tool.ruff]
+exclude = ["data/"]
```

Comment:
Lots of Python code in there, why would we want to skip formatting it?

Reply:
It looks like there is a lot of code related to Jupyter notebooks and I don't want to mess up with something that could have side effects that I don't understand for now.

I suggest to keep it excluded for now until I can discuss with Paul/Christophe about those files.

Here is the ruff output for this directory FYI.

```
data/bwapi/server.py:111:5: E722 Do not use bare `except`
data/common/delete_methods.py:3:8: F401 [*] `bw2io` imported but unused
data/common/distances/distances_to_csv.py:72:41: F821 Undefined name `pycountry`
data/common/distances/distances_to_csv.py:76:18: F821 Undefined name `pycountry`
data/common/distances/distances_to_csv.py:77:22: F821 Undefined name `_`
data/common/distances/transports.py:4:27: F401 [*] `common.export.export_json` imported but unused
data/common/impacts.py:92:8: E721 Do not compare types, use `isinstance()`
data/common/import_.py:95:12: E721 Do not compare types, use `isinstance()`
data/common/import_.py:137:16: E721 Do not compare types, use `isinstance()`
data/common/import_.py:193:15: F541 [*] f-string without any placeholders
data/food/export.py:32:5: F401 [*] `food.ecosystemic_services.ecosystemic_services.plot_ecs_transformations` imported but unused
data/food/export.py:199:51: E721 Do not compare types, use `isinstance()`
data/food/export.py:201:43: E721 Do not compare types, use `isinstance()`
data/food/export.py:204:12: E721 Do not compare types, use `isinstance()`
data/import_agribalyse.py:132:5: F811 Redefinition of unused `sync_datapackages` from line 14
data/import_method.py:12:5: F401 [*] `bw2io.strategies.fix_localized_water_flows` imported but unused
data/import_method.py:15:5: F401 [*] `bw2io.strategies.migrate_exchanges` imported but unused
data/import_method.py:16:5: F401 [*] `bw2io.strategies.normalize_biosphere_categories` imported but unused
data/notebooks/explore.py:6:1: E402 Module level import not at top of file
data/notebooks/explore.py:7:1: E402 Module level import not at top of file
data/notebooks/explore.py:9:1: E402 Module level import not at top of file
data/notebooks/explore.py:12:1: E402 Module level import not at top of file
data/notebooks/explore.py:15:1: E402 Module level import not at top of file
data/notebooks/explore.py:16:1: E402 Module level import not at top of file
data/notebooks/explore.py:17:1: E402 Module level import not at top of file
data/notebooks/explore.py:18:1: E402 Module level import not at top of file
data/notebooks/explore.py:19:1: E402 Module level import not at top of file
data/notebooks/explore.py:21:1: E402 Module level import not at top of file
data/notebooks/explore.py:22:1: E402 Module level import not at top of file
data/notebooks/explore.py:23:1: E402 Module level import not at top of file
data/notebooks/explore.py:24:1: E402 Module level import not at top of file
data/notebooks/explore.py:25:1: E402 Module level import not at top of file
data/notebooks/explore.py:26:1: E402 Module level import not at top of file
data/notebooks/explore.py:27:1: E402 Module level import not at top of file
data/notebooks/explore.py:194:52: E721 Do not compare types, use `isinstance()`
data/notebooks/explore.py:202:15: E741 Ambiguous variable name: `l`
data/notebooks/explore.py:208:88: E721 Do not compare types, use `isinstance()`
data/notebooks/explore.py:373:22: F541 [*] f-string without any placeholders
data/notebooks/explore.py:448:22: F541 [*] f-string without any placeholders
data/notebooks/explore.py:491:22: F541 [*] f-string without any placeholders
data/notebooks/explore.py:601:29: F841 [*] Local variable `e` is assigned to but never used
data/notebooks/ingredients.py:6:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:7:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:11:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:14:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:15:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:16:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:17:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:19:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:20:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:21:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:22:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:23:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:24:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:25:1: E402 Module level import not at top of file
data/notebooks/ingredients.py:472:5: F821 Undefined name `display`
data/notebooks/ingredients.py:488:9: F821 Undefined name `display`
data/notebooks/ingredients.py:665:9: F821 Undefined name `display`
data/notebooks/ingredients.py:674:9: F821 Undefined name `display`
data/notebooks/ingredients.py:705:9: F821 Undefined name `display`
data/notebooks/ingredients.py:711:9: F821 Undefined name `display`
data/notebooks/ingredients.py:722:9: F821 Undefined name `display`
data/notebooks/ingredients.py:733:9: F821 Undefined name `display`
data/notebooks/ingredients.py:744:9: F821 Undefined name `display`
data/notebooks/ingredients.py:753:9: F821 Undefined name `display`
data/notebooks/ingredients.py:759:9: F821 Undefined name `display`
data/notebooks/ingredients.py:769:5: F821 Undefined name `display`
data/notebooks/ingredients.py:801:5: F821 Undefined name `display`
data/notebooks/ingredients.py:812:9: F821 Undefined name `display`
data/notebooks/ingredients.py:820:9: F821 Undefined name `display`
data/notebooks/ingredients.py:827:9: F821 Undefined name `display`
data/notebooks/ingredients.py:856:9: F821 Undefined name `display`
data/notebooks/ingredients.py:859:5: F821 Undefined name `display`
data/notebooks/ingredients.py:867:9: F821 Undefined name `display`
data/notebooks/ingredients.py:876:9: F821 Undefined name `display`
data/notebooks/ingredients.py:893:9: F821 Undefined name `display`
data/notebooks/ingredients.py:903:9: F821 Undefined name `display`
data/notebooks/ingredients.py:913:9: F821 Undefined name `display`
data/notebooks/ingredients.py:918:9: F821 Undefined name `display`
data/notebooks/ingredients.py:938:1: F821 Undefined name `display`
data/notebooks/ingredients.py:1268:34: F541 [*] f-string without any placeholders
data/notebooks/textile_processes.py:74:1: F821 Undefined name `display`
data/spapi/simapro.py:59:13: E722 Do not use bare `except`
data/textile/activities.py:75:36: E721 Do not compare types, use `isinstance()`
data/textile/export.py:32:12: E721 Do not compare types, use `isinstance()`
```